cmd/getStorageValue_test.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/makerdao/vulcanizedb/libraries/shared/factories/storage"
+	"github.com/makerdao/vulcanizedb/libraries/shared/mocks"
+	"github.com/makerdao/vulcanizedb/libraries/shared/storage/types"
+	"github.com/makerdao/vulcanizedb/libraries/shared/transformer"
+	"github.com/makerdao/vulcanizedb/pkg/core"
+	"github.com/makerdao/vulcanizedb/pkg/fakes"
+	"github.com/makerdao/vulcanizedb/test_config"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("getStorageValue Command", func() {
+	var (
+		bc                             *fakes.MockBlockChain
+		db                             = test_config.NewTestDB(test_config.NewTestNode())
+		keysLookupOne, keysLookupTwo   mocks.MockStorageKeysLookup
+		runner                         cmd.StorageValueCommandRunner
+		initializerOne, initializerTwo transformer.StorageTransformerInitializer
+		initializers                   []transformer.StorageTransformerInitializer
+		keyOne, keyTwo                 common.Hash
+		valueOne, valueTwo             common.Hash
+		addressOne, addressTwo         common.Address
+		blockNumber                    int64
+		bigIntBlockNumber              *big.Int
+		fakeHeader                     core.Header
+		headerRepo                     fakes.MockHeaderRepository
+		diffRepo                       mocks.MockStorageDiffRepository
+	)
+
+	BeforeEach(func() {
+		bc = fakes.NewMockBlockChain()
+
+		keysLookupOne = mocks.MockStorageKeysLookup{}
+		keyOne = common.Hash{1, 2, 3}
+		addressOne = fakes.FakeAddress
+		keysLookupOne.SetKeysToReturn([]common.Hash{keyOne})
+		valueOne = common.BytesToHash([]byte{7, 8, 9})
+		bc.SetStorageValuesToReturn(addressOne, valueOne[:])
+
+		keysLookupTwo = mocks.MockStorageKeysLookup{}
+		keyTwo = common.Hash{4, 5, 6}
+		addressTwo = fakes.AnotherFakeAddress
+		keysLookupTwo.SetKeysToReturn([]common.Hash{keyTwo})
+		valueTwo = common.BytesToHash([]byte{10, 11, 12})
+		bc.SetStorageValuesToReturn(addressTwo, valueTwo[:])
+
+		initializerOne = storage.Transformer{
+			Address:           addressOne,
+			StorageKeysLookup: &keysLookupOne,
+			Repository:        &mocks.MockStorageRepository{},
+		}.NewTransformer
+
+		initializerTwo = storage.Transformer{
+			Address:           addressTwo,
+			StorageKeysLookup: &keysLookupTwo,
+			Repository:        &mocks.MockStorageRepository{},
+		}.NewTransformer
+
+		initializers = []transformer.StorageTransformerInitializer{initializerOne, initializerTwo}
+		blockNumber = rand.Int63()
+		bigIntBlockNumber = big.NewInt(blockNumber)
+
+		runner = cmd.NewStorageValueCommandRunner(bc, db, initializers, blockNumber)
+
+		diffRepo = mocks.MockStorageDiffRepository{}
+		runner.StorageDiffRepo = &diffRepo
+
+		headerRepo = fakes.MockHeaderRepository{}
+		fakeHeader = fakes.FakeHeader
+		fakeHeader.BlockNumber = blockNumber
+		headerRepo.GetHeaderReturnHash = fakeHeader.Hash
+		runner.HeaderRepo = &headerRepo
+	})
+
+	AfterEach(func() {
+		test_config.CleanTestDB(db)
```

Comment:
Why do this in an `AfterEach` rather than `BeforeEach`?

Also, do these tests actually need a `db`, or could we maybe get away with passing `nil`?

Reply:
nice catch!